packages/salesforcedx-utils-vscode/src/services/telemetry.ts

```diff
@@ -383,4 +385,60 @@ export class TelemetryService {
         .catch(err => console.error(err));
     }
   }
+
+  private getRandomUserId = (): string => randomBytes(20).toString('hex');
+
+  private buildCliTelemetryCommand(): Command {
+    return new SfCommandBuilder()
+      .withArg('telemetry')
+      .withJson()
+      .build();
+  }
+
+  private async executeCliTelemetry(
+    command: Command
+  ): Promise<string> {
+    const workspacepath = workspaceUtils.getRootWorkspacePath();
+    const execution = new CliCommandExecutor(command, { cwd: workspacepath }
+    ).execute();
+    const cmdOutput = new CommandOutput();
+    const result = cmdOutput.getCmdResult(execution);
+    return result;
+  }
+
```

Comment:
So it was bugging me that we had all these new imports and function that seemed out of the _concern_ of the telemetry module.  I tried creating a UserService class to handle it.  Check out my branch (off your branch) and see what you think. 
gbockus/W-16086212-suggestUserService